packages/firestore/test/unit/specs/query_spec.test.ts

```diff
@@ -98,4 +98,41 @@ describeSpec('Queries:', [], () => {
         hasPendingWrites: true
       });
   });
+
+  specTest(
+    'Ensure correct query results for latency-compensated updates',
+    [],
+    () => {
+      const fullQuery = Query.atPath(ResourcePath.fromString('collection'));
+      const filteredQuery = fullQuery.addFilter(filter('match', '==', true));
+      const docA = doc('collection/a', 1000, { match: false });
+      const docAv2Local = doc(
+        'collection/a',
+        1000,
+        { match: true },
+        { hasLocalMutations: true }
+      );
+
+      return (
+        spec()
+          .userListens(fullQuery)
+          .watchAcksFull(fullQuery, 1000, docA)
+          .expectEvents(fullQuery, { added: [docA] })
+
```

Comment:
I believe that if you do `userUnlisten(fullQuery)` here then you can remove the event right below and further narrow down your test case.

Reply:
I am myself no longer fully sold on my idea. But let me explain what I had in mind:

```
    spec()
          .withGcEnabled(false)
          .userListens(fullQuery)
          .watchAcksFull(fullQuery, 1000, docA)
          .expectEvents(fullQuery, { added: [docA] })
          .userUnlistens(fullQuery)
          .userPatches('collection/a', { match: true })
          .userListens(filteredQuery)
          .expectEvents(filteredQuery, {
            added: [docAv2Local],
            fromCache: true,
            hasPendingWrites: true
          })
```

I believe this would work and it would remove the event in line 124 that is not directly related to your change. The reason I am no longer fully sold on this is that you would have to run the test with GC disabled to make this work. Otherwise, the SDK would delete `docA` from the RemoteDocumentCache as the user unlistens.

Reply:
Yep, can confirm what you described. Without GC disabled this is not working.

I am keep it as-is as that is slightly more likely what happens in real world.